packages/discussions/src/posts.ts

```diff
@@ -0,0 +1,193 @@
+  INestPagination,
+  IPostDTO,
+  INestDeleteResult
+} from "./types";
+
+/**
+ * search posts
+ *
+ * @export
+ * @param {(IQueryPostsOptions | IQueryChannelPostsOptions)} options
+ * @return {*}  {Promise<INestPagination<IPostDTO>>}
+ */
+export function searchPosts(
+  options: IQueryPostsOptions | IQueryChannelPostsOptions
+): Promise<INestPagination<IPostDTO>> {
+  let url = `/posts`;
+  if (options.params.hasOwnProperty("channelId")) {
+    const {
+      params: { channelId }
+    } = options as IQueryChannelPostsOptions;
+    url = `/channels/${channelId}` + url;
+  }
+  options.method = "GET";
+  return request(url, options);
+}
+
+/**
+ * create post
+ *
+ * @export
+ * @param {(ICreatePostOptions | ICreateChannelPostOptions)} options
+ * @return {*}  {Promise<IPostDTO>}
+ */
+export function createPost(
+  options: ICreatePostOptions | ICreateChannelPostOptions
+): Promise<IPostDTO> {
+  let url = `/posts`;
+  if (options.params.hasOwnProperty("channelId")) {
+    const {
+      params: { channelId }
+    } = options as ICreateChannelPostOptions;
+    url = `/channels/${channelId}` + url;
+  }
+  options.method = "POST";
+  return request(url, options);
+}
+
+/**
+ * create reply to post
+ *
+ * @export
+ * @param {(ICreateReplyOptions | ICreateChannelReplyOptions)} options
+ * @return {*}  {Promise<IPostDTO>}
+ */
+export function createReply(
+  options: ICreateReplyOptions | ICreateChannelReplyOptions
+): Promise<IPostDTO> {
+  let url = `/posts/${options.params.postId}/reply`;
+  if (options.params.hasOwnProperty("channelId")) {
+    const {
+      params: { channelId }
+    } = options as ICreateChannelReplyOptions;
+    url = `/channels/${channelId}` + url;
+  }
+  options.method = "POST";
+  return request(url, options);
+}
+
+/**
+ * find post
+ *
+ * @export
+ * @param {(IFindPostOptions | IFindChannelPostOptions)} options
+ * @return {*}  {Promise<IPostDTO>}
+ */
+export function findPost(
+  options: IFindPostOptions | IFindChannelPostOptions
+): Promise<IPostDTO> {
+  let url = `/posts/${options.params.postId}`;
+  if (options.params.hasOwnProperty("channelId")) {
+    const {
+      params: { channelId }
+    } = options as IFindChannelPostOptions;
+    url = `/channels/${channelId}` + url;
+  }
+  options.method = "GET";
+  return request(url, options);
+}
+
+/**
+ * delete post
+ *
+ * @export
+ * @param {(IDeletePostOptions | IDeleteChannelPostOptions)} options
+ * @return {*}  {Promise<INestDeleteResult>}
+ */
+export function deletePost(
+  options: IDeletePostOptions | IDeleteChannelPostOptions
+): Promise<INestDeleteResult> {
+  let url = `/posts/${options.params.postId}`;
+  if (options.params.hasOwnProperty("channelId")) {
+    const {
+      params: { channelId }
+    } = options as IDeleteChannelPostOptions;
+    url = `/channels/${channelId}` + url;
+  }
+  options.method = "DELETE";
+  return request(url, options);
+}
+
+/**
+ * update post
+ * NOTE: this method only updates a post's title and/or body
```

Comment:
This is a good limit to note.

what are your plans for adding update support to geometry or other post attributes?

Reply:
There are separate methods for updating a post's status (only allowed by moderators) and its channel/sharing. This method could support geometry and discussion/target in just a few LOC; it would be beneficial to discuss the business logic about which attributes are updatable and by whom.